### PR TITLE
state mismatch test

### PR DESCRIPTION
**Pull request by cgardens**
_this is not meant to be merged. just a test to demonstrate the issue_

## What
* the current state persistence is: In the db we have the state table with a column called state and then it in that column is a json blob with a single key that is also called state. 
* StatePersistence doesn't store the states with wrapped in that state field.

This was caught when I tried to run the acceptance tests after exposing the new, richer state object in the API

This is _not_ a bug yet, because the `StatePersistence` isn't used anywhere, but we might need to figure out something before we use it.

My hunch is that we need to run a migration to take any states already written to the db and unwrap the them from the state field that is stored in the db.
e.g.
object currently stored in the `state` column
```
{ "state": "{"ads": {}, "ad_sets": {} } }
```
I think we want it to migrate to be this?
```
{"ads": {}, "ad_sets": {} } 
```

heads up @gosusnp

### CODE REVIEW

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceTest.java [hunk 1]
```diff
@@ -82,6 +100,109 @@ void tearDown() throws Exception {
...
+    configRepository.writeStandardSourceDefinition(actorDef);
+    configRepository.writeStandardWorkspace(workspace1);
+    configRepository.writeSourceConnectionNoSecrets(sourceConnection1);
+    configRepository.writeDestinationConnectionNoSecrets(destinationConnection1);
+    // connectionId 1
+    configRepository.writeStandardSync(standardSync1);
+    configRepository.updateConnectionState(connectionId1, state);
+    final Optional<State> state1 = configRepository.getConnectionState(connectionId1);
+    final Optional<StateWrapper> state2 = statePersistence.getCurrentState(connectionId1);
+    assertEquals(state1.get().getState(), state2.get().getLegacyState().get("state"));
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceTest.java:196]: the extra `get("state")` doesn't seem right. if this were the only issue then the `StateWrapper` could just unwrap the extra `state` field, but see the next comment...

File: airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceTest.java [hunk 2]
```diff
@@ -82,6 +100,109 @@ void tearDown() throws Exception {
...
+
+    // connectionId 2
+    configRepository.writeStandardSync(standardSync2);
+    statePersistence.updateOrCreateState(connectionId2, new StateWrapper().withStateType(StateType.LEGACY).withLegacyState(stateJson));
+    final Optional<State> state3 = configRepository.getConnectionState(connectionId2);
+    final Optional<StateWrapper> state4 = statePersistence.getCurrentState(connectionId2);
+    assertEquals(state3.get().getState(), state4.get().getLegacyState());
```
Comment (cgardens) [inline airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceTest.java:203]: `state3.get().getState()` returns null

**Comment by benmoriceau** [comment]
@cgardens @gosusnp Shouldn't we just store (and) the legacy as a `State` object instead of a migration?

**Comment by cgardens** [comment]
Yeah. Agreed. @gosusnp convinced me of this offline too. I think we should create an issue to eventually get rid of the useless wrapper, but no point to mess with a migration now.

**Pull request closed without merging**